feat: add corrupt cookie header monitoring

robrap opened the pull request:
## Description

In case of unusual cookie headers containing "Cookie ",
add custom attributes for monitoring:
- cookies.header.corrupt_count
- cookies.header.corrupt_key_count

See annotation documentation for more details.

Separately, updated to skip cookie log sampling for
0 size cookie header.

## Supporting information

[CR-4614](https://openedx.atlassian.net/browse/CR-4614)

robrap committed:
feat: add cookies.header.is_corrupt monitoring

In case of unusual cookie headers, add custom attribute
cookies.header.is_corrupt for tracking. See its
annotation documentation for more details.

timmc-edx commented on openedx/core/lib/request_utils.py:
```diff
@@ -143,6 +143,14 @@ def log_and_monitor_cookies(self, request):
         # .. custom_attribute_description: The total size in bytes of the cookie header.
         set_custom_attribute('cookies.header.size', cookie_header_size)
 
+        if 'Cookie:' in raw_header_cookie and any('Cookie: ' in key for key in request.COOKIES.keys()):
```
Probably want a trailing space in that first string.

But... how would you feel about recording count information as well? I have a theory that header stuttering could be causing cookie header inflation. But it would also give us more information in case of false positives.

Recording these two as separate metrics would also give us a differential so that we could know if `Cookie: ` was appearing in values, without having to check for it. This would give us some idea of cookie value corruption too.

Something like this:

- `raw_header_cookie.count('Cookie: ')`
- `sum(1 for key in request.COOKIES.keys() if 'Cookie: ' in key)` or whatever the right idiom is in Python

(but I guess only if nonzero)

timmc-edx replied:
That said, I'll go ahead and approve as is, because this by itself would give us more info than we have.

robrap replied:
I’ll try to update tomorrow, because I also want to skip sampling logging if the header is empty.

robrap replied:
Done and done.

robrap committed:
feat: add corrupt cookie header monitoring

In case of unusual cookie headers containing "Cookie ",
add custom attributes for monitoring:
- cookies.header.corrupt_count
- cookies.header.corrupt_key_count

See annotation documentation for more details.

Separately, updated to skip cookie log sampling for
0 size cookie header.

ARCHBOM-2055
